src/theme/index.tsx

```diff
@@ -0,0 +1,47 @@
+export * from './styles'
+export * from './types'
+export * from './utils'
+
+import React, { useMemo } from 'react'
+import { DefaultTheme, ThemeProvider as StyledComponentsThemeProvider } from 'styled-components'
+
+import { getThemePalette } from './utils'
+import { useThemeMode } from 'hooks/useThemeManager'
+
+// Extension/override of styled-components' ThemeProvider but with our own constructed theme object
+const ThemeProvider: React.FC<{ componentKey: Partial<DefaultTheme['componentKey']> }> = ({
```

Comment:
I don't think this is the best way to use the theme.
In my view it should be added to the top of the app hierarchy, dependent only on the global theme state.

The way it's used here every component wanting to use the theme would have to use this provider. Or am I reading this wrong?

Reply:
@alfetopito it will be - right now this is simply extending the default `ThemeProvider` in case we want to use nested `ThemeProvider`s for interpolating over the entire theme, sth like: 
```js
// Define our `fg` and `bg` on the theme
const theme = {
  fg: "palevioletred",
  bg: "white"
};

// This theme swaps `fg` and `bg`
const invertTheme = ({ fg, bg }) => ({
  fg: bg,
  bg: fg
});

render(
  <ThemeProvider theme={theme}>
    <div>
      <Button>Default Theme</Button>

      <ThemeProvider theme={invertTheme}>
        <Button>Inverted Theme</Button>
      </ThemeProvider>
    </div>
  </ThemeProvider>
);
```

the global `ThemeProvider` will wrap below `StaticGlobalStyles` as `ThemedGlobalStyles` and will allow for interpolating theme logic

Reply:
Ook then. I trust you on this